thorlcr/activities/nsplitter/thnsplitterslave.cpp

```diff
@@ -178,16 +193,19 @@ class NSplitterSlaveActivity : public CSlaveActivity, implements ISharedSmartBuf
         if (!inputPrepared)
         {
             inputPrepared = true;
+            activeOutputCount = initActiveOutputCount;
             PARENT::start();
-            remainingOutputs = activeOutputs;
             ForEachItemIn(o, outputs)
             {
                 CSplitterOutput *output = (CSplitterOutput *)outputs.item(o);
-                if (output && output->isStopped())
-                    --remainingOutputs;
+                if (output->isStopped())
+                {
+                    --activeOutputCount;
+                    --stoppedOutputs; // have now considered this output. When stoppedOutputs==activetOutputs in stop(), it will stops the input.
```

Comment:
why not compare stoppedOutputs with initActiveOutputCount (or rename to connectedOutputCount) and avoid the decrement - and some of the race condition?  It would also be clearer since stoppedOutputs would have a consistent meaning.